dist/index.js

```diff
@@ -0,0 +1,166 @@
+
+  try {
+    let res = await fetch(
+      'https://api.dictionaryapi.dev/api/v2/entries/en/' + word,
+    )
+
+    if (res.ok == false) {
+      skeleton.style.display = "none"
+      error.style.display = "flex"
+      wordSection.style.display = "block"
+      meaningsDiv.style.display = "none"
+      meaningDiv.style.display = "none"
+      return false
+    } else {
+        skeleton.style.display = "none"
+        error.style.display = "none"
+        wordSection.style.display = "block"
+        meaningsDiv.style.display = "block"
+      meaningDiv.style.display = "block"
+      return await res.json()
+    }
+  } catch (error) {}
+}
+
+function playAudio() {
+  audio.play()
+}
+
+async function handleSubmit(e) {
+  e.preventDefault()
+  if (wordInput.value.length <= 1) {
+    errorLable.style.display = 'flex'
+    return
+  } else {
+    errorLable.style.display = 'none'
+  }
+
+  const data = await fecthWord(wordInput.value)
+  if (data == false) {
+    return
+  }
+
+  let getPhoneticText = data[0]?.phonetics.find((item, index) => {
+    if (item.text?.length > 0) return true
+  })
+
+  let getPhoneticAudio = data[0]?.phonetics.find((item, index) => {
+    if (item.audio?.length > 0) return true
+  })
+
+  audio.setAttribute('src', getPhoneticAudio.audio)
+
+  word.textContent = wordInput.value
+  phoneticText.textContent = getPhoneticText.text
+
+  let meaningsString = ''
+  data[0]?.meanings.map((item) => {
+    meaningsString += meaningsHtml(item)
+  })
+  meaningsDiv.innerHTML = meaningsString
+}
+
+function meaningsHtml(meaning) {
+  let others = ' '
+  meaning.definitions.map((item) => (others += othersHtml(item)))
+  let html = `<div class="block">
+                 <header class="text-sm md:text-base font-semibold capitalize">${meaning.partOfSpeech}</header>
+                 <ul class=" list-decimal text-sm md:text-base px-5">
+                 ${others}
+                 </ul>
+                 </div>`
+  
+  html = html.trim()
+  return html
+}
+
+function othersHtml(definition) {
+  definitionHtml = `<li  class="leading-5 p-2"> ${definition.definition}<ul>`
+  let exampleHtml = ''
+  let synonymsHtml = ''
+  let antonymsHtml = ''
+  exampleHtml +=
+    definition.example == undefined
+      ? ' '
+      : `<li class="italic font-normal text-muted">sentence: “${definition.example}"</li>`
+  synonymsHtml +=
+    definition.synonyms == undefined
+      ? ' '
+      : `<li class="font-medium">Synonyms: “${flatArray(
+          definition.synonyms,
+        )}”</li>`
+  antonymsHtml +=
+    definition.antonyms == undefined
+      ? ' '
+      : `<li class="font-medium">Synonyms: “${flatArray(
+          definition.antonyms,
+        )}”</li>`
+
+  definitionHtml += exampleHtml + synonymsHtml + antonymsHtml + '</ul></li>'
+  return definitionHtml
+}
+
+function flatArray(arr) {
+  let word = ''
+  arr.map((item) => (word = word + item + ', '))
+  let i = word.lastIndexOf(', ')
+  word = word.substring(0, i)
+  return word.trim()
```

Comment:
arr.reduce() can accomplish the same thing without the need for line 130 and 131

Reply:
Try it out

Reply:
Absolutelty will do, reduce will work better.